src/libraries/Common/tests/System/IO/VirtualDriveHelper.Windows.cs

```diff
@@ -0,0 +1,160 @@
+                Directory.Delete(VirtualDriveTargetDir, recursive: true);
+            }
+        }
+        catch { } // avoid exceptions on dispose
+    }
+
+    /// <summary>
+    /// Returns the path of a folder that is to be mounted using SUBST.
+    /// </summary>
+    public string VirtualDriveTargetDir
+    {
+        get
+        {
+            if (_virtualDriveTargetDir == null)
+            {
+                // Create a folder inside the temp directory so that it can be mounted to a drive letter with subst
+                _virtualDriveTargetDir = Path.Join(Path.GetTempPath(), Path.GetRandomFileName());
+                Directory.CreateDirectory(_virtualDriveTargetDir);
+            }
+
+            return _virtualDriveTargetDir;
+        }
+    }
+
+    /// <summary>
+    /// Returns the drive letter of a drive letter that represents a mounted folder using SUBST.
+    /// </summary>
+    public char VirtualDriveLetter
+    {
+        get
+        {
+            if (_virtualDriveLetter == default)
+            {
+                // Mount the folder to a drive letter
+                _virtualDriveLetter = CreateVirtualDrive(VirtualDriveTargetDir);
+            }
+            return _virtualDriveLetter;
+        }
+    }
+
+    ///<summary>
+    /// On Windows, mounts a folder to an assigned virtual drive letter using the subst command.
+    /// subst is not available in Windows Nano.
+    /// </summary>
+    public char CreateVirtualDrive(string targetDir)
+    {
+        if (!OperatingSystem.IsWindows())
+        {
+            throw new PlatformNotSupportedException();
+        }
+
+        char driveLetter = GetNextAvailableDriveLetter();
+        bool success = RunProcess(CreateProcessStartInfo("cmd", "/c", SubstPath, $"{driveLetter}:", targetDir));
+        if (!success || !DriveInfo.GetDrives().Any(x => x.Name[0] == driveLetter))
+        {
+            throw new InvalidOperationException($"Could not create virtual drive {driveLetter}: with subst");
+        }
+        return driveLetter;
+
+        // Finds the next unused drive letter and returns it.
+        char GetNextAvailableDriveLetter()
+        {
+            List<char> existingDrives = DriveInfo.GetDrives().Select(x => x.Name[0]).ToList();
+
+            // A,B are reserved, C is usually reserved
+            IEnumerable<int> range = Enumerable.Range('D', 'Z' - 'D');
+            IEnumerable<char> castRange = range.Select(x => Convert.ToChar(x));
+            IEnumerable<char> allDrivesLetters = castRange.Except(existingDrives);
+
+            if (!allDrivesLetters.Any())
+            {
+                throw new ArgumentOutOfRangeException("No drive letters available");
+            }
+
+            return allDrivesLetters.First();
+        }
+    }
+
+    /// <summary>
+    /// On Windows, unassigns the specified virtual drive letter from its mounted folder.
+    /// </summary>
+    public void DeleteVirtualDrive(char driveLetter)
```

Comment:
It seems that this method is not used by the test classes and it does not modify the helper state. So we can make it private static.

```suggestion
    private static void DeleteVirtualDrive(char driveLetter)
```